### PR TITLE
nothing really matters?

### PR DESCRIPTION
**Pull request by ChrisRackauckas**

### CODE REVIEW

**Comment by YingboMa** [comment]
Should we adapt https://github.com/JuliaLang/julia/pull/29727 in JuliaDiffEq?

File: src/integrators/integrator_utils.jl [hunk 1]
```diff
@@ -466,10 +484,11 @@ function reset_fsal!(integrator)
...
   # integrator.reeval_fsal = false
+  nothing
 end
 
 function (integrator::ODEIntegrator)(t,deriv::Type=Val{0};idxs=nothing)
   current_interpolant(t,integrator,idxs,deriv)
 end
 
-(integrator::ODEIntegrator)(val::AbstractArray,t::Union{Number,AbstractArray},deriv::Type=Val{0};idxs=nothing) = current_interpolant!(val,t,integrator,idxs,deriv)
+(integrator::ODEIntegrator)(val::AbstractArray,t::Union{Number,AbstractArray},deriv::Type=Val{0};idxs=nothing) = (current_interpolant!(val,t,integrator,idxs,deriv); nothing)
```
Comment (YingboMa) [inline src/integrators/integrator_utils.jl]: This change might be breaking. Also, in Base, mutating functions generally return something, e.g.
```julia
julia> reverse!(collect(1:3))
3-element Array{Int64,1}:
 3
 2
 1

julia> push!(ones(2), 1)
3-element Array{Float64,1}:
 1.0
 1.0
 1.0
```

**Comment by ChrisRackauckas** [comment]
>Should we adapt JuliaLang/julia#29727 in JuliaDiffEq?

This was discussed at triage. Making sure to return `nothing` will be the recommendation, but not necessarily `return nothing`.

File: src/integrators/integrator_utils.jl (hunk 1 above)
Comment (ChrisRackauckas) [reply src/integrators/integrator_utils.jl]: something non-deterministically broke a split methods test. I don't think it's this though.

Comment (ChrisRackauckas) [reply src/integrators/integrator_utils.jl]: Sure, we can drop this nothing

**Comment by ChrisRackauckas** [comment]
I was doing this manually, so if you have a better method then that would be helpful

**Comment by ChrisRackauckas** [comment]
FWIW test timings don't improve from this, so maybe perform_step! would be more helpful.

**Pull request closed without merging**